htlcswitch/control_tower.go

```diff
@@ -106,51 +101,41 @@ func (p *paymentControl) ClearForTakeoff(htlc *lnwire.UpdateAddHTLC) error {
 			)
 
 		case channeldb.StatusInFlight:
-			// We already have an InFlight payment on the network. We will
-			// disallow any more payment until a response is received.
-			takeoffErr = ErrPaymentInFlight
+			// We already have an InFlight payment on the network.
+			// We will disallow any more payment until a response
+			// is received.
+			return ErrPaymentInFlight
 
 		case channeldb.StatusCompleted:
-			// We've already completed a payment to this payment hash,
-			// forbid the switch from sending another.
-			takeoffErr = ErrAlreadyPaid
+			// We've already completed a payment to this payment
+			// hash, forbid the switch from sending another.
+			return ErrAlreadyPaid
 
 		default:
-			takeoffErr = ErrUnknownPaymentStatus
```

Comment:
This commit should be reverted, and leave the batch error handling as is. If the function returns an error, then each will be run in isolation. This is the intent behind capturing the errors, but allowing the transaction to return `nil`

https://github.com/etcd-io/bbolt/blob/master/db.go#L787

Reply:
Ah, I see TIL! Will revert and maybe add a comment explaining this 👍

Reply:
I was kinda puzzled by why it was done this way tbh, probably should have asked before changing it :p